modules/thermal_hydraulics/src/components/HeatStructureCylindrical.C

```diff
@@ -37,7 +37,8 @@ HeatStructureCylindrical::HeatStructureCylindrical(const InputParameters & param
   for (unsigned int i = 0; i < _names.size(); i++)
     _name_index[_names[i]] = i;
 
-  _material_names = getParam<std::vector<std::string>>("materials");
+  _material_names = isParamValid("materials") ? getParam<std::vector<std::string>>("materials")
+                                              : std::vector<std::string>{};
```

Comment:
same

Reply:
Can not used default because `checkEqualSize` will fail. The reason for it to fail is "name" is a required param. It doesn't has default value. Then the size won't be equal

Reply:
use the size check there instead of isParamValid

Reply:
I prefer not to change it. Both way works in my opinion.

Reply:
I dont have a strong opinion there. They both work indeed and it s quite subjective